scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt

```diff
@@ -0,0 +1,413 @@
+  }
+  return finalUpdatedList
+}
+
+/** Retrieves the list of [MavenDependency] from maven_dependencies.textproto. */
+private fun retrieveMavenDependencyList(): List<MavenDependency> {
+  return getProto(
+    "maven_dependencies.pb",
+    MavenDependencyList.getDefaultInstance()
+  ).mavenDependencyList.toList()
+}
+
+/**
+ * Helper function to parse the textproto file to a proto class.
+ *
+ * @param textProtoFileName name of the textproto file to be parsed
+ * @param proto instance of the proto class
+ * @return proto class from the parsed textproto file
+ */
+private fun getProto(
+  textProtoFileName: String,
+  proto: MavenDependencyList
+): MavenDependencyList {
+  val protoBinaryFile = File("scripts/assets/$textProtoFileName")
+  val builder = proto.newBuilderForType()
+  val protoObject = FileInputStream(protoBinaryFile).use {
+    builder.mergeFrom(it)
+  }.build() as MavenDependencyList
+  return protoObject
+}
+
+private fun parseArtifactName(artifactName: String): String {
+  var colonIndex = artifactName.length - 1
+  while (artifactName.isNotEmpty() && artifactName[colonIndex] != ':') {
+    colonIndex--
+  }
+  val artifactNameWithoutVersion = artifactName.substring(0, colonIndex)
+  val parsedArtifactNameBuilder = StringBuilder()
+  for (index in artifactNameWithoutVersion.indices) {
+    if (artifactNameWithoutVersion[index] == '.' || artifactNameWithoutVersion[index] == ':' ||
+      artifactNameWithoutVersion[index] == '-'
+    ) {
+      parsedArtifactNameBuilder.append(
+        '_'
+      )
+    } else {
+      parsedArtifactNameBuilder.append(artifactNameWithoutVersion[index])
+    }
+  }
+  return parsedArtifactNameBuilder.toString()
+}
+
+private fun runBazelQueryCommand(rootPath: String): List<String> {
+  val rootDirectory = File(rootPath).absoluteFile
+  val bazelClient = BazelClient(rootDirectory)
+  val bazelQueryDepsNames = mutableListOf<String>()
+
+  val output = bazelClient.retrieveThirdPartyMavenDependenciesList()
+  output.forEach { dep ->
+    bazelQueryDepsNames.add(dep.substring(MAVEN_PREFIX_LENGTH, dep.length))
+  }
+  bazelQueryDepsNames.sort()
+  return bazelQueryDepsNames.toList()
+}
+
+private fun readMavenInstall(
+  pathToMavenInstall: String,
+  bazelQueryDepsNames: List<String>
+): List<MavenListDependency> {
+  val mavenInstallJson = File(pathToMavenInstall)
+  val mavenInstallJsonText = mavenInstallJson.inputStream().bufferedReader().use { it.readText() }
+  val moshi = Moshi.Builder().addLast(KotlinJsonAdapterFactory()).build()
+  val adapter = moshi.adapter(MavenListDependencyTree::class.java)
+  val dependencyTree = adapter.fromJson(mavenInstallJsonText)
+  val mavenInstallDependencyList = dependencyTree?.mavenListDependencies?.dependencyList
+  val finalDependenciesList = mutableListOf<MavenListDependency>()
+  mavenInstallDependencyList?.forEach { dep ->
+    val artifactName = dep.coord
+    val parsedArtifactName = parseArtifactName(artifactName)
```

Comment:
What does 'parsed' mean here? Prefer trying to contextualize what the value specifically represents in its name.

Reply:
Removed the variable.

Reply:
Verified